superset-frontend/plugins/plugin-chart-table/src/controlPanel.tsx

```diff
@@ -191,35 +188,41 @@ const config: ControlPanelConfig = {
           },
         ],
         [
-          hasGenericChartAxes && isAggMode
-            ? {
-                name: 'time_grain_sqla',
-                config: {
-                  ...sharedControls.time_grain_sqla,
-                  visibility: ({ controls }) => {
-                    const dttmLookup = Object.fromEntries(
-                      ensureIsArray(controls?.groupby?.options).map(option => [
-                        option.column_name,
-                        option.is_dttm,
-                      ]),
-                    );
+          {
+            name: 'time_grain_sqla',
+            config: {
+              ...sharedControls.time_grain_sqla,
+              visibility: ({ controls }) => {
```

Comment:
Not related to this PR. These are bunch of repeated logics that we should probably centralize at some point

Reply:
True. I also noticed many repeated code between ECharts plugins. I'm delaying removing this repetition until after the migrations because I think we'll have a better overview of the common patterns.